protocols/raft/src/test/java/io/atomix/protocols/raft/zeebe/ZeebeTest.java

```diff
@@ -0,0 +1,203 @@
+      stopwatch.stop();
+    }
+
+    logger.info("Test run time: {}", stopwatch.toString());
+    stop();
+  }
+
+  private Collection<ZeebeTestNode> buildNodes() {
+    return nodeSuppliers.stream()
+        .map(supplier -> supplier.apply(temporaryFolder))
+        .collect(Collectors.toList());
+  }
+
+  @SuppressWarnings("squid:S2699") // awaitAllContain is the assert here
+  @Test
+  public void shouldAppendAndReplicate() {
+    // given
+    final int partitionId = 1;
+    final ZeebeLogAppender appender = helper.awaitLeaderAppender(partitionId);
+    final byte[] data = new byte[] {1};
+
+    // when
+    final Indexed<ZeebeEntry> appended = appender.appendEntry(data).join();
+
+    // then
+    helper.awaitAllContain(partitionId, appended);
+  }
+
+  @Test
+  public void shouldNotCompactAnything() {
+    // given
+    final int partitionId = 1;
+    final RaftPartitionServer server = helper.awaitLeaderServer(1);
+    final ZeebeLogAppender appender = helper.awaitLeaderAppender(partitionId);
+    final byte[] data = new byte[] {1};
+
+    // when
+    final Indexed<ZeebeEntry> firstAppended = appender.appendEntry(data).join();
+    for (int i = 0; i < ENTRIES_PER_SEGMENT; i++) {
+      helper.awaitAllContain(partitionId, appender.appendEntry(data).join());
+    }
+    server.snapshot().join();
+
+    // then
+    assertTrue(helper.containsIndexed(server, firstAppended));
+  }
+
+  @Test
+  public void shouldCompactUpToCompactablePosition() {
+    // given
+    final int partitionId = 1;
+    final RaftPartitionServer server = helper.awaitLeaderServer(1);
+    final ZeebeLogAppender appender = helper.awaitLeaderAppender(partitionId);
+    final byte[] data = new byte[] {1};
+
+    // when
+    Indexed<ZeebeEntry> appended = appender.appendEntry(data).join();
+    final Indexed<ZeebeEntry> firstAppended = appended;
+    for (int i = 0; i < ENTRIES_PER_SEGMENT; i++) {
+      appended = appender.appendEntry(data).join();
+      helper.awaitAllContain(partitionId, appended);
+    }
+    server.setCompactablePosition(appended.index(), appended.entry().term());
+    server.snapshot().join();
+
+    // then
+    assertFalse(helper.containsIndexed(server, firstAppended));
+    assertTrue(helper.containsIndexed(server, appended));
+  }
+
+  @Test
+  public void shouldAppendAndReplicateOnFailOver() {
+    // given
+    final int partitionId = 1;
+    final ZeebeTestNode originalLeader = helper.awaitLeader(partitionId);
+    final byte[] data = new byte[] {1};
+
+    // when
+    final Indexed<ZeebeEntry> firstEntry =
+        helper.awaitLeaderAppender(partitionId).appendEntry(data).join();
+    helper.awaitContains(originalLeader, partitionId, firstEntry);
+    originalLeader.stop().join();
+
+    originalLeader.start(nodes).join();
+    final Indexed<ZeebeEntry> secondEntry =
+        helper.awaitLeaderAppender(partitionId).appendEntry(data).join();
+
+    // then
+    helper.awaitAllContain(partitionId, secondEntry);
```

Comment:
In this case it can happen that the first entry is gone right?

Reply:
It's possible it wasn't appended everywhere, true. Tbh I wasn't sure about the value of this test, it's more a make-sure-we-didn't-break-anything kind of test, since that functionality is normally tested in the Raft tests.

Reply:
I added a second check to ensure the first entry is present everywhere.

Reply:
It could be that the entry is not replicated everywhere before we shutdown the leader, true, then the test could be flaky with the second check. Like I said, I'm not sure of the value of the test in the long run. It did make me realise that the RaftPartitionGroupConfig is serialized though so that was useful (since it's the only test where we stop/start a node). A more useful test might be rather writing constantly to the leader and restarting followers, and making sure at the end all entries are there eventually (but not shutting down the leader).